lambdas/data-migration1/tests/test-rules.js

```diff
@@ -0,0 +1,394 @@
+
+  rulesModel = new Rule();
+  await rulesModel.createTable();
+  await createBucket(process.env.system_bucket);
+
+  fakeCollection = fakeCollectionFactory();
+  fakeProvider = fakeProviderFactory({
+    encrypted: false,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  });
+
+  t.context.knexAdmin = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      migrationDir,
+    },
+  });
+  await t.context.knexAdmin.raw(`create database "${testDbName}";`);
+  await t.context.knexAdmin.raw(`grant all privileges on database "${testDbName}" to "${testDbUser}"`);
+
+  t.context.knex = await getKnexClient({
+    env: {
+      ...localStackConnectionEnv,
+      PG_DATABASE: testDbName,
+      migrationDir,
+    },
+  });
+
+  await t.context.knex.migrate.latest();
+
+  await Promise.all([
+    putJsonS3Object(
+      process.env.system_bucket,
+      messageTemplateKey,
+      { meta: 'meta' }
+    ),
+    putJsonS3Object(
+      process.env.system_bucket,
+      workflowfile,
+      { testworkflow: 'workflow-config' }
+    ),
+  ]);
+});
+
+test.afterEach.always(async (t) => {
+  await t.context.knex('rules').del();
+  await t.context.knex('providers').del();
+  await t.context.knex('collections').del();
+});
+
+test.after.always(async (t) => {
+  await providersModel.deleteTable();
+  await collectionsModel.deleteTable();
+  await rulesModel.deleteTable();
+
+  await recursivelyDeleteS3Bucket(process.env.system_bucket);
+
+  await t.context.knex.destroy();
+  await t.context.knexAdmin.raw(`drop database if exists "${testDbName}"`);
+  await t.context.knexAdmin.destroy();
+});
+
+test.serial('migrateRuleRecord correctly migrates rule record', async (t) => {
+  const { knex, providerKmsKeyId } = t.context;
+  fakeCollection = fakeCollectionFactory();
+  fakeProvider = fakeProviderFactory({
+    encrypted: false,
+    privateKey: 'key',
+    cmKeyId: 'key-id',
+    certificateUri: 'uri',
+    createdAt: Date.now(),
+    updatedAt: Date.now(),
+  });
+  const fakeRule = generateFakeRule(fakeCollection.name, fakeCollection.version, fakeProvider.id);
+
+  await migrateCollectionRecord(fakeCollection, knex);
+  await migrateProviderRecord(fakeProvider, providerKmsKeyId, knex);
+  await migrateRuleRecord(fakeRule, knex);
+
+  const createdRecord = await t.context.knex.queryBuilder()
+    .select()
+    .table('rules')
+    .where({ name: fakeRule.name })
+    .first();
+
+  t.deepEqual(
+    omit(createdRecord, ['cumulus_id', 'collection_cumulus_id', 'provider_cumulus_id']),
+    omit(
+      {
+        ...fakeRule,
+        arn: fakeRule.rule.arn ? fakeRule.rule.arn : null,
+        type: fakeRule.rule.type,
+        value: fakeRule.rule.value ? fakeRule.rule.value : null,
+        enabled: true,
+        log_event_arn: null,
+        execution_name_prefix: null,
+        payload: null,
+        queue_url: null,
+        tags: null,
```

Comment:
I think I'd prefer to have the test validating the successful migration record including values for all possible fields. Then leave the nullable testing for that separate test.

I'd also like this test to verify that the correct provider/collection are attached to the rule. You could create those records in RDS before doing the migration.